lib/jsonWebToken.js

```diff
@@ -178,17 +179,17 @@ exports.verify = function(jwtString, PEMKey, options, callback) {
   if (typeof payload.exp !== 'number')
     return done(new Error('invalid exp value in payload'));
   if (!options.ignoreExpiration) {
-    if (Math.floor(Date.now() / 1000) >= payload.exp) {
+    if (Math.floor(Date.now() / 1000) >= payload.exp + constants.CLOCK_SKEW) {
       return done(new Error('jwt is expired'));
     }
   }
 
   // (5) nbf
   //   - check if it exists
   if (payload.nbf) {
-    if (typeof payload.nbf !== 'number')
+    if (typeof payload.nbf !== 'number' || payload.nbf >= payload.exp)
```

Comment:
It would be helpful to users if the error message indicated that nbf > exp OR nbf was not a number